website/docs/cookbooks/introduction/formatting.md

```diff
@@ -0,0 +1,174 @@
+```bash
+scala-cli fmt --check .
+```
+
+This command will check if the code is formatted correctly according to the default rules.
+If it is not, it will return a diff of the changes that need to be made to match formatting rules.
+
+### Format
+
+If you want to format all your code, you can use the following command:
+```bash
+scala-cli fmt .
+```
+This command will format all your sources in the current directory according to the default rules.
+
+### Example
+
+Let's take a simple `HelloWorld` application:
+```scala title=HelloWorld.scala
+// use scala 3.3.0
+@main def hello()={println("Hello, World!")}
+def someLongMethodMoreThan80Characters():Unit={println("This is a long method")
+println("Another print here?")}
+```
+
+You can notice that this implementations is written with long lines, no indentation, no spaces and no new lines.
+It is hard to distinguish where one statement ends and another begins, which makes it hard to read.
+
+<ChainedSnippets>
+
+Let's check the format of this code:
+```bash
+scala-cli fmt --check .
+```
+Here is the output:
+```diff
+--- a/private/tmp/formatting-example/FormattingExample.scala
++++ b/private/tmp/formatting-example/FormattingExample.scala
+@@ -1,4 +1,7 @@
+ // use scala 3.3.0
+-@main def hello()={println("Hello, World!")}
+-def someLongMethodMoreThan80Characters():Unit={println("This is a long method")
+-println("Another print here?")}
++@main def hello() = { println("Hello, World!") }
++def someLongMethodMoreThan80Characters(): Unit = {
++  println("This is a long method")
++  println("Another print here?")
++}
++
+error: --test failed
+```
+</ChainedSnippets>
+
+As we can see, the code is not formatted by default rules. And [Scalafmt](https://scalameta.org/scalafmt/) formatter provides a diff of the changes that need to be made to match formatting rules.
+
+<ChainedSnippets>
+To apply suggested changes to our code, we can run:
+
+```bash
+scala-cli fmt .
+```
+And there will be no output, which means that the formatting was successful. Let's see the code after formatting:
+
+```scala title=HelloWorld.scala
+// use scala 3.3.0
+@main def hello() = { println("Hello, World!") }
+def someLongMethodMoreThan80Characters(): Unit = {
+  println("This is a long method")
+  println("Another print here?")
+}
+```
+</ChainedSnippets>
+
+Now the code is formatted according to the default rules.
+
+<ChainedSnippets>
+Let's check it again:
+
+```bash
+scala-cli fmt --check .
+```
+
+And now we seee `All files are formatted with scalafmt :)` message, which means that the code is formatted correctly.
+
+</ChainedSnippets>
+
+---
+
+## Advanced usage
+
+This section describes how to use Scala CLI to format your code using custom parameters and `.scalafmt.conf` file.
+
+
+### Default 
+By default, Scala CLI creates a `.scalafmt.conf` file in the `.scala-build/.scalafmt.conf` directory with default rules. The default rules are:
+```hocon
+version = 3.7.12
+runner.dialect = scala3
+```
+
```

Comment:
```suggestion

:::note
The default version of `scalafmt` is linked to the Scala CLI version you are using. In case you want to use a different version, you can override it with the `--fmt-version` option. The defaults might also change after doing a Scala CLI version update.
:::
```